community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -789,6 +790,53 @@ public int copyTo( int sourceOffset, PageCursor targetCursor, int targetOffset,
         return 0;
     }
 
+    @Override
+    public int copyTo( int sourceOffset, ByteBuffer buf )
+    {
+        if ( buf.getClass() == UnsafeUtil.directByteBufferClass )
```

Comment:
I think buf.isReadOnly() should be checked not only for `UnsafeUtil.directByteBufferClass` but for all buffers that provided; then you will not gonna need  `assert isDirectWritableByteBuffer( buf )` since that will be duplicated check.
Also `buf.getClass() == UnsafeUtil.directByteBufferClass` implies that byteBuffer classloader and UnsafeUtilClass loader is the same, maybe we need to mention it at least as a comment.

Reply:
I can move the checks into the if-condition, sending callers with weird byte buffers into the fall-back code.

They are JDK classes, so they should have the same class loaders, either the system class loader or the boot class loader.